src/ripple/shamap/impl/SHAMapDelta.cpp

```diff
@@ -123,12 +123,12 @@ SHAMap::compare (std::shared_ptr<SHAMap> const& otherMap,
 
     assert (isValid () && otherMap && otherMap->isValid ());
 
-    using StackEntry = std::pair <SHAMapTreeNode*, SHAMapTreeNode*>;
-    std::stack <StackEntry, std::vector<StackEntry>> nodeStack; // track nodes we've pushed
-
     if (getHash () == otherMap->getHash ())
         return true;
 
+    using StackEntry = std::pair <SHAMapTreeNode*, SHAMapTreeNode*>;
```

Comment:
They aren't the same:

    using StackEntry = std::pair <SHAMapTreeNode*, SHAMapTreeNode*>;

and

    using StackEntry = std::shared_ptr<SHAMapTreeNode>;